functions/bigquery.js

```diff
@@ -0,0 +1,170 @@
+const functions = require('firebase-functions');
+
+const _ = require('lodash');
+
+const { BigQuery } = require('@google-cloud/bigquery');
+
+const {
+  INBOUND_TABLE,
+  INTAKE_TABLE,
+  REIMBURSEMENTS_TABLE,
+  getAllRecords
+} = require('./airtable');
+
+const bigQueryClient = new BigQuery({
+  projectId: functions.config().bigquery.project_id,
+  credentials: {
+    private_key: functions.config().bigquery.credentials.private_key.split('\\n').join('\n'),
+    client_email: functions.config().bigquery.credentials.client_email,
+  }
+});
+
+const OPS_DATASET_ID = 'operations';
+
+const INBOUND_TYPES = {
+  id: 'STRING',
+  status: 'STRING',
+  intakeTime: 'TIMESTAMP',
+  method: 'STRING',
+};
+
+const INTAKE_TYPES = {
+  id: 'STRING',
+  dateCreated: 'TIMESTAMP',
+  dueDate: 'TIMESTAMP',
+  costCategory: 'INTEGER',
+  householdSize: 'INTEGER',
+  items: 'STRING',
+  neighborhood: 'STRING',
+  status: 'STRING',
+  ticketID: 'STRING',
+  timeline: 'STRING',
+  vulnerability: 'STRING',
+};
+
+const REIMBURSEMENT_TYPES = {
+  id: 'STRING',
+  ticketRecords: 'STRING',
+  totalCost: 'FLOAT',
+  donation: 'FLOAT',
+  netReimbursement: 'FLOAT',
+  // TODO : get rid of this nonesense
+  fundMgr: 'STRING',
+  dateSubmitted: 'TIMESTAMP',
+  paymentPlatform: 'STRING',
+};
+
+const TABLE_TO_TYPES = {
+  [INBOUND_TABLE]: INBOUND_TYPES,
+  [INTAKE_TABLE]: INTAKE_TYPES,
+  [REIMBURSEMENTS_TABLE]: REIMBURSEMENT_TYPES,
+};
+
+function _convertRecords(table) {
+  return ([id, fields]) => {
+    if (!_.has(TABLE_TO_TYPES, table)) {
+      throw Error('Invalid table', { table: table });
+    }
+
+    return _.assign(
+      { id: id },
+      // Filter out all fields with null values
+      _.pickBy(
+        // Map fields in the record to fields in the bigquery type mappings
+        _.mapValues(
+          TABLE_TO_TYPES[table],
+          (type, fieldName) => {
+            let val = fields[fieldName];
+
+            if (!val) {
+              return null;
+            }
+
+            // Validate types and values
+            if (type === 'STRING' || type === 'TIMESTAMP') {
```

Comment:
i would maybe break this whole validation method out entirely just for readability